traitsui/qt4/check_list_editor.py

```diff
@@ -132,11 +151,42 @@ def update_editor(self):
             pass
 
 
-class CustomEditor(SimpleEditor):
+class CustomEditor(BaseCheckListEditor):
     """ Custom style of editor for checklists, which displays a set of check
         boxes.
     """
 
+    def init(self, parent):
+        """ Finishes initializing the editor by creating the underlying toolkit
+            widget.
+        """
+        self.create_control(parent)
+        super(CustomEditor, self).init(parent)
+
+    def dispose(self):
+        """ Disposes of the contents of an editor.
+        """
+        if self.control is not None:
+            self.clear_layout()
+
+        if self._mapper is not None:
+            self._mapper.mapped[str].disconnect(self.update_object)
+            self._mapper = None
+
+        super().dispose()
+
+    def dispose_child(self, child):
+        """ Carry out necessary clean up before a child widget is removed from
+        the control layout.
+
+        Parameters
+        ----------
+        child : QWidget
+        """
+        if self._mapper is not None:
+            child.clicked.disconnect(self._mapper.map)
+            self._mapper.removeMappings(child)
+
```

Comment:
An alternative to this is to maintain a list of signals inside `CheckListEditor`, like in #877. I might try that now...

Reply:
See alternative PR: https://github.com/enthought/traitsui/pull/882

Reply:
One thing I did not like about `dispose_child` is that it is only useful if all the children have the same type and the same function such that the same disposal code can be applied to everyone of them. This is unlikely to be generally useful: A layout can have a mixture of widgets.
(Preferring #882 to this one)